IaC/scripts/create-cluster-env.bash

```diff
@@ -0,0 +1,398 @@
+
+# Sane prog env:
+set -o errexit -o pipefail -o noclobber # -o nounset
+aze() { echo "executing: az $@";az "$@"; }
+dockere() { echo "executing: docker $@";docker "$@"; }
+kubectle() { echo "executing: kubectl $@";kubectl "$@"; }
+# Functions
+usage(){
+    echo """
+    $0 --ngsa-prefix basename123 [Optional Args/Flags]
+    $0 -s azure-subs -n basename123 [Optional Args/Flags]
+
+Required args:
+    -n | --ngsa-prefix BASE_NAME    This will be the NGSA prefix for all resources
+                                    Do not include punctuation - only use a-z and 0-9
+                                    must be at least 5 characters long
+                                    must start with a-z (only lowercase)
+Optional args:
+    -s | --subscription AZ_SUB      Azure Subscription Name or ID
+    -e | --env ENVIRONMENT          Environemnt Type. Default: dev (See README.md for other values)
+    -d | --domain DOMAIN_NAME       Registered Domain Name. Default: nip.io (Requires --email)
+    -m | --email EMAIL_DOMAIN       Required Email if a '--domain' is given
+    -l | --location LOCATION        Location where the resources will be created. Default: westus2
+    -k | --k8s-ver K8S_VERSION      Kubernetes version used. Default: 1.18.8
+                                    Use 'az aks get-versions -l westus2 -o table' to get supported versions
+    -c | --node-count NODE_COUNT    Cluster Node Count. Default: 3
+    -r | --dns-rg DNS_RG            DNS Resource group name. Default: dns-rg
+    -i | --cosmos-key COSMOS_KEY    
+    -u | --cosmos-url COSMOS_URL    In case users want to use their own CosmosDBBoth Key and URL are empty by default. 
+Optional Flag:
+    -x | --set-k8s-context          Sets the kubernetes context for current user in $HOME/.kube/config
+    -o | --smoke-it                 Set it to true or false to enable smokers deployment. Default: false
+    -h | --help                     Show the usage
+"""
+}
+echoc(){
+    c=$1;shift;b=$(tput bold);r=$(tput setaf 1);g=$(tput setaf 2);b=$(tput setaf 4);y=$(tput setaf 3);d=$(tput sgr0)
+    echo -e $(eval "echo \$${c}")"$@""${d}"
+}
+echo_exit(){
+    ec=$1;shift;echo -e "\n$@" >&2
+    exit $ec
+}
+
+[[ "$#" == 0 ]] && usage && exit 0
+while (( "$#" )); do
+    key="$1" val="$2"
+    case "$key" in
+        -s|--subscription)
+        AZ_Sub=$val
+        shift 2
+        ;;
+
+        -n|--ngsa-prefix)
+        Ngsa_Base_Name=$val
+        shift 2
+        ;;
+
+        -e|--env)
+        Ngsa_Env=$val
+        shift 2
+        ;;
+
+        -d|--domain)
+        Ngsa_Domain_Name=$val
+        shift 2
+        ;;
+
+        -m|--email)
+        Ngsa_Email=$val
+        shift 2
+        ;;
+
+        -k|--k8s-ver)
+        Ngsa_K8S_VER=$val
+        shift 2
+        ;;
+
+        -c|--node-count)
+        Ngsa_Node_Count=$val
+        shift 2
+        ;;
+
+        -r|--dns-rg)
+        Ngsa_DNS_RG=$val
+        shift 2
+        ;;
+
+        -o|--smoke-it)
+        Ngsa_Smoke=yes
```

Comment:
Should we calls this Ngsa_Loderunner or Ngsa_l8r?

Reply:
Yes, I'm changing it in the AKS-Script PR. Will change it here afterwards